### PR TITLE
Small convenience updates

### PR DESCRIPTION
**Pull request by sergey-unikie**
Few files added
- .gitignore to avoid accidental pushing any binary into the repo
- .clang-format - For now, this is only (preliminary) placeholder, but we may add automatic formatting later
- update_dependencies.sh  - convenient way to install/update dependencies locally

### CODE REVIEW

**Comment by sergey-unikie** [comment]
Would it make sense to remove binary .deb files from fog_sw/tools? They could be automatically downloaded and installed in the script. In this way we can also control package versions much easier, than by replacing deb file.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,19 @@
+ros2_ws/build
+ros2_ws/install
+ros2_ws/log
+packaging/*.deb
+
+# needed for CLion compatibility
+.idea
+CMakeLists.txt
```
Comment (maseabunikie) [inline .gitignore:8]: Should this file be kept visible to git changes?

Comment (sergey-unikie) [reply .gitignore:8]: there is no CMakeLists.txt in the root repo, and also `colcon build` don't need it.
I locally have my own CMakeLists.txt for integration with CLion IDE.
So this line just for myself :) I don't want accidentally push it

**Comment by sergey-unikie** [comment]
@jnippula   we may also put list of ubuntu dependencies into separate file and use some tricks to install them. 
https://askubuntu.com/questions/252734/apt-get-mass-install-packages-from-a-file

In this way we would only need 1 single file with such list, Dockerfile and local script may both use it

File: .gitignore (hunk 1 above)
Comment (maseabunikie) [reply .gitignore:8]: Ok. Clear.

**Comment by maseabunikie** [comment]
@sergey-unikie 
We can also have a debian meta package(s) which do not contain any files. They just point to needed dependencies.

**Comment by jnippula** [comment]
> @jnippula we may also put list of ubuntu dependencies into separate file and use some tricks to install them.
> https://askubuntu.com/questions/252734/apt-get-mass-install-packages-from-a-file
> 
> In this way we would only need 1 single file with such list, Dockerfile and local script may both use it

Yes, this is good. The dependencies could also contain the version numbers of modules, to verify that we have a consistent env.

And we should have separate build and run dep lists. Also, as Jari R. has mentioned we should consider having separate run deps for each vm running on hypervisor.

**Review by kulmesa — APPROVED** [review]

**Pull request merged**